### PR TITLE
Prepare release 5.8.0

### PR DESCRIPTION
**Pull request by adidahiya**
#### CHANGELOG.md entry:

[no-log]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,107 @@
...
+- [bugfix] Removed warning printed to console when using the [`no-unused-variable`](https://palantir.github.io/tslint/rules/no-unused-variable/) along with the `noUnusedLocals` and `noUnusedParameters` compiler options (#3227)
+- [bugfix] [`no-invalid-this`](https://palantir.github.io/tslint/rules/no-invalid-this/) ignores functions with a `this` param (#3267)
+- [enhancement] Sort failures by line and character for formatters (#3345)
+- [enhancement] [`import-blacklist`](https://palantir.github.io/tslint/rules/import-blacklist/) also checks exports and dynamic imports (#3258)
+- [enhancement] [`no-conditional-assignment`](https://palantir.github.io/tslint/rules/no-conditional-assignment/) added check for conditional (ternary) expressions (#2629)
+- [enhancement] Allow [`completed-docs`](https://palantir.github.io/tslint/rules/completed-docs/) to list doc tags that mark a node as not requiring a documentation body. Tags can also provide a regexp matcher to validate that their contents are docs-valid. (#2415)
+- [enhancement] [`await-promise`](https://palantir.github.io/tslint/rules/await-promise/) enforces that `for-await-of` is only used with `AsyncIterableIterator` (#3297)
```
Comment (ajafff) [inline CHANGELOG.md]: s/AsyncIterableIterator/AsyncIterable/

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,107 @@
...
+- [enhancement] [`one-line`](https://palantir.github.io/tslint/rules/one-line/) checks type alias declarations (#3200)
+- [enhancement] [`deprecation`](https://palantir.github.io/tslint/rules/deprecation/) checks object destructuring (#3318)
+- [enhancement] [`no-submodule-imports`](https://palantir.github.io/tslint/rules/no-submodule-imports/) also checks exports (#3258)
+- [enhancement] [`restrict-plus-operands`](https://palantir.github.io/tslint/rules/restrict-plus-operands/): More specific error message when arguments include strings (#3220)
+- [enhancement] [`no-unsafe-any`](https://palantir.github.io/tslint/rules/no-unsafe-any/) checks more expressions, for example destructuring, `yield`, property initializer (#3196)
+- [enhancement] [`object-literal-sort-keys`](https://palantir.github.io/tslint/rules/object-literal-sort-keys/): allow grouping of object properties via additional blank lines when using alphabetical ord
+ering. (#3191)
```
Comment (ajafff) [inline CHANGELOG.md]: is this line break intended?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,107 @@
...
+- [enhancement] Migrated CLI from using `colors` module to `chalk` module (#3171)
+- [enhancement] [`no-unused-variable`](https://palantir.github.io/tslint/rules/no-unused-variable/) apples the ignorePattern to imports (#3187)
```
Comment (ajafff) [inline CHANGELOG.md]: appl**i**es

**Comment by ajafff** [comment]
Should also include #3362

File: CHANGELOG.md (hunk 2 above)
Comment (adidahiya) [reply CHANGELOG.md]: nope, good catch

**Pull request merged**